packages/pages/src/common/src/feature/stream.test.ts

```diff
@@ -45,3 +52,101 @@ describe("stream", () => {
     expect(streamConfig).toEqual(expectedStreamConfig);
   });
 });
+
+const siteStreamPath = "foo/bar";
+
+describe("formatSiteStream", () => {
+  it("errors and exits when there are multiple entityIds", () => {
+    const testJson = { filter: { entityIds: ["1234", "123"] } };
+    const mockExit = vi
+      .spyOn(process, "exit")
+      .mockImplementation(() => undefined as never);
+    formatSiteStream(testJson, siteStreamPath);
+    expect(mockExit).toHaveBeenCalledWith(1);
+  });
+
+  it("returns expected entityId", () => {
+    const testJson = { filter: { entityIds: ["1234"] } };
+    const expectedJson = { entityId: "1234" };
+    expect(formatSiteStream(testJson, siteStreamPath)).toEqual(expectedJson);
+  });
+
+  it("returns expected id with id first", () => {
+    const testJson = { fields: ["meta", "name"], $id: "123" };
+    const expectedJson = { id: "123", fields: ["meta", "name"] };
+    expect(formatSiteStream(testJson, siteStreamPath)).toEqual(expectedJson);
+  });
+
+  it("returns expected full config", () => {
+    const testJson = {
+      $id: "123",
+      fields: ["meta", "name"],
+      filter: { entityIds: ["1234"] },
+      source: "foo",
+      localization: ["en"],
+    };
+    const expectedJson = {
+      id: "123",
+      entityId: "1234",
+      fields: ["meta", "name"],
+      localization: ["en"],
+    };
+    expect(formatSiteStream(testJson, siteStreamPath)).toEqual(expectedJson);
+  });
+});
+
+describe("readSiteStream", () => {
+  const projectStructure = new ProjectStructure({});
+
+  it("reads siteStream from config.yaml", () => {
+    const path = "config.yaml";
+    fs.writeFileSync(
+      path,
+      `siteStream:
+        id: site-stream
+        entityId: site-stream-from-yaml
+        fields:
+          - c_visualLayouts.c_visualConfiguration
+        localization:
+          locales:
+            - en
+      `
+    );
+    const siteStream = readSiteStream(projectStructure);
+    expect(siteStream).toEqual({
+      id: "site-stream",
+      entityId: "site-stream-from-yaml",
+      fields: ["c_visualLayouts.c_visualConfiguration"],
+      localization: { locales: ["en"] },
+    });
+    fs.rmSync(path);
```

Comment:
If the tests fail will this still delete the file? I'm guessing no and we need to do some sort of after call or try/catch